esp/logging/loggingmanager/loggingmanager.cpp

```diff
@@ -242,6 +259,70 @@ bool CLoggingManager::updateLog(IEspContext* espContext, IEspUpdateLogRequestWra
     return bRet;
 }
 
+bool CLoggingManager::saveToTankFile(IEspUpdateLogRequestWrap& logRequest, CLogRequestInFile* reqInFile)
+{
+    if (!logFailSafe.get())
+    {
+        ERRLOG("CLoggingManager::saveToTankFile: logFailSafe not configured.");
+        return false;
+    }
+
+    unsigned startTime = (getEspLogLevel()>=LogNormal) ? msTick() : 0;
+
+    StringBuffer GUID;
+    logFailSafe->GenerateGUID(GUID, NULL);
+    reqInFile->setGUID(GUID);
+    reqInFile->setOption(logRequest.getOption());
+
+    StringBuffer reqBuf;
+    Owned<IEspUpdateLogRequestWrap> logRequestFiltered = logContentFilter.filterLogContent(&logRequest);
+    if (!serializeLogRequestContent(logRequestFiltered, GUID, reqBuf))
+    {
+        ERRLOG("CLoggingManager::saveToTankFile: failed in serializeLogRequestContent().");
+        return false;
+    }
+
+    logFailSafe->AddACK(GUID);//Ack this logging request since the task will be done as soon as the next line is called.
+    logFailSafe->Add(GUID, reqBuf, reqInFile);
+
+    ESPLOG(LogNormal, "LThread:saveToTankFile: %dms\n", msTick() - startTime);
+    return true;
+}
+
+unsigned CLoggingManager::serializeLogRequestContent(IEspUpdateLogRequestWrap* request, const char* GUID, StringBuffer& logData)
+{
+    addXMLItemToBuf(LOGREQUEST_GUID, GUID, logData);
+
+    const char* option = request->getOption();
+    if (!isEmptyString(option))
+        addXMLItemToBuf(LOGREQUEST_OPTION, option, logData);
+
+    logData.append("<").append(LOGREQUEST).append(">");
+
+    const char* logRequest = request->getUpdateLogRequest();
+    MemoryBuffer memBuf;
+    LZWCompress(logRequest, strlen(logRequest), memBuf, 0x100);
+    JBASE64_Encode(memBuf.toByteArray(), memBuf.length(), logData);
+
+    logData.append("</").append(LOGREQUEST).append(">");
+
+    return logData.length();
+}
+
+void CLoggingManager::addXMLItemToBuf(const char* tag, int value, StringBuffer& buf)
+{
+    buf.append("<").append(tag).append(">");
+    buf.append(value);
+    buf.append("</").append(tag).append(">");
+}
+
+void CLoggingManager::addXMLItemToBuf(const char* tag, const char* value, StringBuffer& buf)
```

Comment:
Do you know that value does not need to be xml encoded?

Reply:
No need for the calls in this PR. But, I will add an option just in case of future use.